pkg/providers/instance/instance.go

```diff
@@ -196,7 +198,7 @@ func (p *Provider) createAKSIdentifyingExtension(ctx context.Context, vmName str
 	return nil
 }
 
-func (p *Provider) newNetworkInterfaceForVM(vmName string, backendPools *loadbalancer.BackendAddressPools, instanceType *corecloudprovider.InstanceType) armnetwork.Interface {
+func (p *Provider) newNetworkInterfaceForVM(ctx context.Context, vmName string, backendPools *loadbalancer.BackendAddressPools, instanceType *corecloudprovider.InstanceType) armnetwork.Interface {
```

Comment:
a lot of our code is littered with this ctx object pattern. I wonder if some sort of global configuration options defined an accessible anywhere would be better.